core/src/main/java/hudson/Functions.java

```diff
@@ -1062,6 +1062,10 @@ public static String getFooterURL() {
             Descriptor d = c.getInstance();
             if (d.getGlobalConfigPage()==null)  continue;
 
+            if (!Jenkins.get().hasPermission(d.getPermission())) {
+                continue;
+            }
+
```

Comment:
This takes the place of the `DescriptorVisibilityFilter` in an earlier attempt.

The problem there was that we either had to allow any descriptor that wasn't a `GlobalConfiguration` (which leaves all `DescriptorImpl/global.jelly` accessible, which is unsafe), or we actually filtered out descriptors from, well, everywhere:

![Screenshot](https://user-images.githubusercontent.com/1831569/74558503-b200af00-4f62-11ea-9fdb-b2e5eefd3538.png)

While funny, this wasn't useful.